extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js

```diff
@@ -168,6 +188,30 @@ export class GoogleSubscriptionsPlatform {
    * Perdforms the pingback to the subscription platform
    */
   pingback() {}
+
+  /**
+   * @param {!../../../src/service/viewer-impl.Viewer} viewer
+   * @private
+   */
+  resolveGoogleViewer_(viewer) {
+    // This is a very light veiwer resolution since there's no real security
+    // implication - this only affects on-platform preferences.
+    const viewerUrl = viewer.getParam('viewerUrl');
+    if (viewerUrl) {
```

Comment:
why do we need 2 different blocks?
cant we just call `viewer.getViewerOrigin().then(origin => {` and assume that is resolves immediately if `viewerUrl` is already available?

Reply:
The comment above explains it. Unfortunately this is a pretty tough situation. The action execution must be synchronous, otherwise we get hit with a popup blocker. Thus I'm trying to resolve one as soon as possible and the other can wait.